crates/storage/api/mod.rs

```diff
@@ -0,0 +1,103 @@
+//! # Storage Backend API
+//!
+//! This module provides a thin, minimal interface for storage backends:
+//!
+//! - Thin: Minimal set of operations that databases must provide
+//! - Simple: Avoids type-system complexity and focuses on core functionality
+//!
+//! Rather than implementing business logic in each database backend, this API
+//! provides low-level primitives that higher-level code can build upon.
+//! This eliminates code duplication and makes adding new database backends trivial.
+//!
+//! The API differentiates between three types of database access:
+//!
+//! - Read-only transactions ([`StorageRoTx`])
+//! - Read-write transactions ([`StorageRwTx`])
+//! - Locked snapshots ([`StorageLocked`]): Persistent read-only views, righ now it's
+//!   only used in snapsync stage.
```

Comment:
We should include this documentation in the `developers` section of the book (`docs/` folder).

Reply:
Nvm. I can add it to this PR